float32 cur_roll
float32 cur_pitch
float32 cur_yaw
float32 cur_distance
